### PR TITLE
Support commits for using Mender C++ client.

### PR DESCRIPTION
**Pull request by kacf**
Note: This includes a merge from master. Please ignore everything before the "Merge branch 'master'" commit near the bottom.

### CODE REVIEW

File: meta-mender-core/recipes-mender/mender-client/mender-client-cpp.inc [hunk 1]
```diff
@@ -87,7 +96,9 @@ FILES:mender-auth += "\
     ${systemd_unitdir}/system/mender-authd.service \
 "
 
-DEPENDS += "lmdb googletest"
+DEPENDS += "lmdb googletest boost libarchive"
+RDEPENDS:mender-auth:append = " boost-filesystem"
```
Comment (oleorhagen) [inline meta-mender-core/recipes-mender/mender-client/mender-client-cpp.inc:100]: Hmm, is this really all the boost stuff we rely on?

File: meta-mender-core/recipes-mender/mender-client/mender-client_git.bb [hunk 2]
```diff
@@ -1,4 +1,4 @@
-require mender-client-go.inc
+require mender-client-cpp.inc
```
Comment (oleorhagen) [inline meta-mender-core/recipes-mender/mender-client/mender-client_git.bb:1]: What's the benefit?

**Review by oleorhagen — APPROVED** [review]
Looks good!

File: meta-mender-core/recipes-mender/mender-client/mender-client-cpp.inc (hunk 1 above)
Comment (kacf) [reply meta-mender-core/recipes-mender/mender-client/mender-client-cpp.inc:100]: Yep!

Remember that `RDEPENDS` is only libraries. The rest of the stuff we use is header-only, a build-time dependency, which is therefore covered by the generic `DEPENDS = "boost"`

Comment (oleorhagen) [reply meta-mender-core/recipes-mender/mender-client/mender-client-cpp.inc:100]: Alright, sweet!

File: meta-mender-core/recipes-mender/mender-client/mender-client_git.bb (hunk 2 above)
Comment (kacf) [reply meta-mender-core/recipes-mender/mender-client/mender-client_git.bb:1]: None! 😄 I messed this up, we should of course be using the `mender_git.bb` recipe instead. I'll remove this commit, thanks for catching.

**Pull request merged**